lib/http/HttpClient_WinInet.cpp

```diff
@@ -339,16 +335,20 @@ class WinInetRequestWrapper
                         // must stay valid and writable until the next
                         // INTERNET_STATUS_REQUEST_COMPLETE callback comes
                         // (that's why those are member variables).
+                        LOG_TRACE("InternetReadFile() failed: ERROR_IO_PENDING. Waiting for INTERNET_STATUS_REQUEST_COMPLETE to be called again");
                         return;
                     }
                     LOG_WARN("InternetReadFile() failed: %d", dwError);
                     break;
                 }
 
                 m_bodyBuffer.insert(m_bodyBuffer.end(), m_buffer, m_buffer + m_bufferUsed);
-            } while (m_bufferUsed == sizeof(m_buffer));
+            } while (m_bufferUsed != 0);
         }
 
+        std::unique_ptr<SimpleHttpResponse> response(new SimpleHttpResponse(m_id));
+        response->m_body = m_bodyBuffer;
+
         // SUCCESS with no IO_PENDING means we're done with the response body: try to parse the response headers.
         if (dwError == ERROR_SUCCESS) {
```

Comment:
Do we need to reset the state of the new members after the InternetReadFile operation completes? (is this class designed to be reused again?)

Reply:
No, it is only made for one request.
It is deleted at the end of this call